src/modules/sudo.rs

```diff
@@ -18,7 +18,15 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         return None;
     }
 
-    let is_sudo_cached = context.exec_cmd("sudo", &["-n", "true"]).is_some();
+    let mut sudo_option = "-nv";
+
+    /* let's check if -N option is supported */
+    let is_sudo_nocreds_update = context.exec_cmd("sudo", &["-NV"]).is_some();
```

Comment:
wouldn't this run sudo twice on each run of starship?

Reply:
Maybe try inspecting the error output?

Reply:
In principle I think we could avoid calling sudo twice on the optimistic path by using `-Nnv`, then falling back to `-nv` only if the first command fails, but I don't see a way to avoid calling it twice in general unless we cache the observation that the system sudo is an old version.

Reply:
I've done some testing on this. As far as I can tell, it is never faster to get the version from sudo than it is to have it barf on an invalid flag. We would additionally need to parse the output (which would be fast, but adds a small amount of additional work) to decide what to do with that output.

I haven't observed a system where parsing the version output would be significantly faster than what this is currently doing. As for the solution I proposed, it could save ~8ms on a ~30ms operation, but only when the sudo version is >= 1.9.12 **and** the user already has cached credentials. In all other cases, the performance would be roughly the same (and the 8ms savings is based on the best-case scenario over all the systems I tried. In most cases, it would be more like 3ms on a 20ms operation).

Given these results, I'm willing to say that the performance issues here are probably fine, especially given that on most systems I've tested, the cost of a `sudo -NV` with a old sudo is somewhere between 0.5-2ms.